### PR TITLE
[WFCORE-5286] Remove deprecated javax.api usage from modules

### PR DESCRIPTION
**Pull request by boris-unckel**
Fixes https://issues.redhat.com/browse/WFCORE-5286

### CODE REVIEW

**Comment by boris-unckel** [comment]
@ropalka This failure (timeout and parser) are not related to the change. Do I miss something? Please review.

**Review by ropalka — APPROVED** [review]
LGTM

**Comment by boris-unckel** [comment]
@jmesnil Review is done. Please check and merge.

File: core-feature-pack/common/src/main/resources/modules/system/layers/base/org/apache/httpcomponents/core/main/module.xml [hunk 1]
```diff
@@ -34,7 +34,9 @@
 
 
     <dependencies>
-        <module name="javax.api"/>
+        <!--WFCORE-5286 Remove deprecated <module name="javax.api"/> -->
```
Comment (darranl) [inline core-feature-pack/common/src/main/resources/modules/system/layers/base/org/apache/httpcomponents/core/main/module.xml:37]: I think these lines should just be removed if we are removing them, we don't send to leave our files with a build up of commented out items unless we really need to keep the removal visible in the future e.g. commenting out a redundant error code to stop it being used again in the future.

Comment (boris-unckel) [reply core-feature-pack/common/src/main/resources/modules/system/layers/base/org/apache/httpcomponents/core/main/module.xml]: @darranl thanks for your review. I have removed the commented line in all cases. Please review again.

File: core-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/as/domain-http-interface/main/module.xml [hunk 2]
```diff
@@ -36,6 +36,7 @@
     </resources>
 
     <dependencies>
+        <module name="java.desktop"/>
```
[the remark this answers is not in the document]
Comment (boris-unckel) [reply core-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/as/domain-http-interface/main/module.xml:39]: @jmesnil thanks for review. 
`jdeps wildfly-domain-http-interface-15.0.0.Final.jar` shows:
`org.jboss.as.domain.http.server                    -> java.beans                                         java.desktop`

[the remark this answers is not in the document]
Comment (jmesnil) [reply core-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/as/domain-http-interface/main/module.xml:39]: I tried jdeps on the master branch and it only shows a dependency to `java.beans`:

```
$ jdeps ~/.m2/repository/org/wildfly/core/wildfly-domain-http-interface/16.0.0.Beta3-SNAPSHOT/wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar
wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar -> not found
wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar -> /Library/Java/JavaVirtualMachines/jdk1.8.0_212.jdk/Contents/Home/jre/lib/rt.jar
   org.jboss.as.domain.http.server (wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar)
      -> io.undertow.attribute                              not found
      -> io.undertow.io                                     not found
      -> io.undertow.predicate                              not found
      -> io.undertow.protocols.ssl                          not found
      -> io.undertow.security.handlers                      not found
      -> io.undertow.server                                 not found
      -> io.undertow.server.handlers                        not found
      -> io.undertow.server.handlers.cache                  not found
      -> io.undertow.server.handlers.encoding               not found
      -> io.undertow.server.handlers.error                  not found
      -> io.undertow.server.handlers.form                   not found
      -> io.undertow.server.handlers.resource               not found
      -> io.undertow.server.protocol.http                   not found
      -> io.undertow.util                                   not found
      -> java.beans                                         
      -> java.io                                            
      -> java.lang                                          
      -> java.lang.invoke                                   
      -> java.net                                           
      -> java.nio                                           
      -> java.nio.charset                                   
      -> java.nio.file                                      
      -> java.security                                      
      -> java.security.cert                                 
      -> java.util                                          
      -> java.util.concurrent                               
      -> java.util.concurrent.atomic                        
      -> java.util.function                                 
      -> java.util.regex                                    
      -> java.util.stream                                   
      -> javax.net.ssl                                      
      -> org.jboss.as.controller                            not found
      -> org.jboss.as.controller.capability                 not found
      -> org.jboss.as.controller.client                     not found
      -> org.jboss.as.controller.descriptions               not found
      -> org.jboss.as.controller.management                 not found
      -> org.jboss.as.controller.remote                     not found
      -> org.jboss.as.core.security                         not found
      -> org.jboss.as.domain.http.server.cors               wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar
      -> org.jboss.as.domain.http.server.logging            wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar
      -> org.jboss.as.domain.http.server.security           wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar
      -> org.jboss.as.domain.management                     not found
      -> org.jboss.as.protocol                              not found
      -> org.jboss.as.protocol.mgmt.support                 not found
      -> org.jboss.dmr                                      not found
      -> org.jboss.modules                                  not found
      -> org.jboss.msc                                      not found
      -> org.jboss.msc.service                              not found
      -> org.wildfly.common                                 not found
      -> org.wildfly.elytron.web.undertow.server            not found
      -> org.wildfly.security.auth.server                   not found
      -> org.wildfly.security.http                          not found
      -> org.wildfly.security.manager                       not found
      -> org.xnio                                           not found
      -> org.xnio.channels                                  not found
      -> org.xnio.conduits                                  not found
      -> org.xnio.ssl                                       not found
      -> org.xnio.streams                                   not found
   org.jboss.as.domain.http.server.cors (wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar)
      -> io.undertow.server                                 not found
      -> io.undertow.server.handlers                        not found
      -> io.undertow.util                                   not found
      -> java.lang                                          
      -> java.util                                          
      -> org.jboss.as.domain.http.server.logging            wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar
   org.jboss.as.domain.http.server.logging (wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar)
      -> java.io                                            
      -> java.lang                                          
      -> java.net                                           
      -> java.util                                          
      -> org.jboss.logging                                  not found
      -> org.jboss.modules                                  not found
   org.jboss.as.domain.http.server.security (wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar)
      -> io.undertow.io                                     not found
      -> io.undertow.security.api                           not found
      -> io.undertow.security.idm                           not found
      -> io.undertow.security.impl                          not found
      -> io.undertow.server                                 not found
      -> io.undertow.server.handlers                        not found
      -> io.undertow.util                                   not found
      -> java.io                                            
      -> java.lang                                          
      -> java.lang.invoke                                   
      -> java.net                                           
      -> java.nio                                           
      -> java.nio.charset                                   
      -> java.security                                      
      -> java.security.cert                                 
      -> java.util                                          
      -> java.util.function                                 
      -> javax.security.auth                                
      -> javax.security.auth.callback                       
      -> javax.security.auth.kerberos                       
      -> javax.security.sasl                                
      -> org.ietf.jgss                                      
      -> org.jboss.as.controller                            not found
      -> org.jboss.as.controller.descriptions               not found
      -> org.jboss.as.controller.security                   not found
      -> org.jboss.as.core.security                         not found
      -> org.jboss.as.domain.http.server                    wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar
      -> org.jboss.as.domain.http.server.logging            wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar
      -> org.jboss.as.domain.management                     not found
      -> org.jboss.dmr                                      not found
      -> org.wildfly.common.iteration                       not found
      -> org.wildfly.security.auth.callback                 not found
      -> org.wildfly.security.auth.server                   not found
      -> org.wildfly.security.credential                    not found
      -> org.wildfly.security.evidence                      not found
      -> org.wildfly.security.password                      not found
      -> org.wildfly.security.password.interfaces           not found

```

could you please depend on `java.beans`?

[the remark this answers is not in the document]
Comment (boris-unckel) [reply core-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/as/domain-http-interface/main/module.xml:39]: @jmesnil This seems to be a misunderstanding. "java.beans" is a package only. According to [https://jboss-modules.github.io/jboss-modules/manual/#predefined-modules](https://jboss-modules.github.io/jboss-modules/manual/#predefined-modules) and [https://github.com/jboss-modules/jboss-modules/blob/1.x/src/main/resources/jdk-module-paths/java.desktop](https://github.com/jboss-modules/jboss-modules/blob/1.x/src/main/resources/jdk-module-paths/java.desktop) for JDK8 and [https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/module-summary.html](https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/module-summary.html) for JDK11 there is only a module "java.desktop" including the package "java.beans".

Please try jdeps with a JDK11 implementation. It does show the modules as summary at the beginning:
`[borisunckel@localhost wildfly-core]$ /usr/lib/jvm/adoptopenjdk-11-hotspot/bin/jdeps ~/.m2/repository/org/wildfly/core/wildfly-domain-http-interface/16.0.0.Beta3-SNAPSHOT/wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar`
```
wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar -> java.base
wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar -> java.desktop
wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar -> java.security.jgss
wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar -> java.security.sasl
wildfly-domain-http-interface-16.0.0.Beta3-SNAPSHOT.jar -> not found
```
and in the details (shortend)
`org.jboss.as.domain.http.server                    -> java.beans                                         java.desktop`

[the remark this answers is not in the document]
Comment (jmesnil) [reply core-feature-pack/common/src/main/resources/modules/system/layers/base/org/jboss/as/domain-http-interface/main/module.xml:39]: thanks for the explanation

**Comment by jmesnil** [comment]
@boris-unckel thanks

**Pull request merged**